app/routes/user/email_notifications.server.ts

```diff
@@ -0,0 +1,208 @@
+  }
+
+  // Create
+  async createEmailNotification(notification: EmailNotification) {
+    if (!notification.name)
+      throw new Response('name must not be empty', { status: 400 })
+    if (!notification.recipient)
+      throw new Response('recipient must not be empty', { status: 400 })
+    if (!notification.topics)
+      throw new Response('topics must not be empty', { status: 400 })
+
+    const created = Date.now()
+    const uuid = crypto.randomUUID()
+
+    const db = await tables()
+    await db.email_notification.put({
+      sub: this.#sub,
+      uuid,
+      name: notification.name,
+      created,
+      topics: notification.topics,
+      active: true,
+      recipient: notification.recipient,
+    })
+
+    // Materialized view
+    for (const topic of notification.topics) {
+      await db.email_notification_subscription.put({
+        uuid,
+        topic,
+        recipient: notification.recipient,
+      })
+    }
+  }
+
+  // Read
+  async getEmailNotifications() {
+    const db = await tables()
+    const results = await db.email_notification.query({
+      KeyConditionExpression: '#sub = :sub',
+      ExpressionAttributeNames: {
+        '#sub': 'sub',
+        '#uuid': 'uuid',
+        '#name': 'name',
+        '#created': 'created',
+        '#topics': 'topics',
+        '#recipient': 'recipient',
+      },
+      ExpressionAttributeValues: {
+        ':sub': this.#sub,
+      },
+      ProjectionExpression: '#uuid, #created, #name, #topics, #recipient',
+    })
+    const emailNotifications = results.Items as EmailNotificationVM[]
+    for (const notice of emailNotifications) {
+      let formats = []
+      notice.noticeTypes = []
+      for (const topic of notice.topics) {
+        let mappedData = mapTopicsToFormatAndNoticeType(topic)
+        formats.push(mappedData.noticeFormat)
+        notice.noticeTypes.push(mappedData.noticeType)
+      }
+      notice.format = [...new Set(formats)][0]
+    }
+    emailNotifications.sort((a, b) => a.created - b.created)
+    return emailNotifications
+  }
+
+  async getEmailNotification(uuid: string): Promise<EmailNotificationVM> {
+    const db = await tables()
+    const item = (await db.email_notification.get({
+      sub: this.#sub,
+      uuid,
+    })) as ({ sub: string } & EmailNotificationVM) | null
+    if (!item) throw new Response(null, { status: 404 })
+    let formats = []
+    item.noticeTypes = []
+    for (const topic of item.topics) {
+      let mappedData = mapTopicsToFormatAndNoticeType(topic)
+      formats.push(mappedData.noticeFormat)
+      item.noticeTypes.push(mappedData.noticeType)
+    }
+    item.format = [...new Set(formats)][0]
+    const { sub, ...notification } = item
+    return {
+      uuid,
+      ...notification,
+    }
+  }
+
+  // Update
+  async updateEmailNotification(email_notification: EmailNotification) {
+    if (!email_notification.uuid) return null
+    const db = await tables()
+    db.email_notification.update(
```

Comment:
```suggestion
    await db.email_notification.update(
```

Reply:
I intentionally left the await out because it gives this message:
![image](https://user-images.githubusercontent.com/11023698/182878686-60ee2a8e-172f-429c-93ad-89ba1aa4a4c8.png)

Reply:
That's because you have a callback. Try removing the callback.

Reply:
That did it!